core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -491,6 +495,19 @@ abstract class AbstractFetcherThread(name: String,
     } finally partitionMapLock.unlock()
   }
 
+  def addTopicIdsToThread(partitions: Set[TopicPartition], topicIds: String => Option[Uuid]) = {
+    partitionMapLock.lockInterruptibly()
+    try {
+      partitions.foreach { tp =>
+        val currentState = partitionStates.stateValue(tp)
```

Comment:
Should we ensure that there is actually a state? It must be there but it might be better to be safe.